File: leetcode/2medium/이경호/396.cpp
class Solution {
public:
  int maxRotateFunction(vector<int>& nums) {
    // F(0)
    int prev = 0;
    for (int i = 0; i < nums.size(); i++) {
      prev += i * nums[i];
    }
    // F(k(> 0))
    int max = prev;
    int numsSum = reduce(nums.begin(), nums.end(), 0);
    for (int k = 1; k < nums.size(); k++) {
      prev += numsSum;
      prev -= (nums.size()) * nums[nums.size() - k];
      max = std::max(max, prev);
    }
    return max;
  }
};

File: leetcode/2medium/오하늘/396.py
from collections import deque
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    def maxRotateFunction(self, nums: List[int]) -> int:
        ans = float('-inf') # 음수의 경우가 있음
        nums_deque = deque(nums)
        n = len(nums)

        for k in range(n):
            nums_deque.rotate(1) # 한 칸씩 회전
            temp = 0
            for i in range(n):
                temp+=i*nums_deque[i]
            if temp>ans:
                ans=temp

        return ans
